pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotTableIdealStateBuilder.java

```diff
@@ -310,10 +310,19 @@ public Boolean call() throws Exception {
       if (bootstrapHosts == null || bootstrapHosts.isEmpty()) {
         throw new RuntimeException("Invalid value for " + Helix.DataSource.Realtime.Kafka.KAFKA_BROKER_LIST);
       }
-      // TODO: Factory should be loaded from config
-      PinotKafkaConsumerFactory _pinotKafkaConsumerFactory = new SimpleConsumerFactory();
-      PinotKafkaConsumer consumerWrapper = _pinotKafkaConsumerFactory.buildMetadataFetcher(bootstrapHosts,
-          PinotTableIdealStateBuilder.class.getSimpleName() + "-" + kafkaTopicName, KAFKA_CONNECTION_TIMEOUT_MILLIS);
+      PinotKafkaConsumerFactory pinotKafkaConsumerFactory = null;
+      Map<String, String> decoderConfigs = null;
+      try {
+        pinotKafkaConsumerFactory = (PinotKafkaConsumerFactory) Class.forName(_kafkaStreamMetadata.getConsumerFactory()).newInstance();
+        LOGGER.info("Setting pinot consumer factory to {}", pinotKafkaConsumerFactory.getClass());
+        decoderConfigs = _kafkaStreamMetadata.getDecoderProperties();
+      } catch (Exception e) {
+          pinotKafkaConsumerFactory = new SimpleConsumerFactory();
+          LOGGER.info("Caught exception getting consumer factory, setting to default {}",
```

Comment:
warn, or maybe even error?

Reply:
The reason this isn't error is because by default, this factory isn't set and defaults to simple. We only set it internally to use a different consumer.